sale_analytic_account/models/sale_order.py

```diff
@@ -0,0 +1,15 @@
+from odoo import models
+
+
+class SaleOrder(models.Model):
+    _inherit = 'sale.order'
+
+    def action_confirm(self):
+        res = super(SaleOrder, self).action_confirm()
+        for so in self:
+            if any(so.order_line.filtered(
+                    lambda line: line.product_id.create_analytic_account)):
```

Comment:
```suggestion
            if so.order_line.filtered('create_analytic_account'):
```
I think this is the same, right?